spec/lib/submission_builder/ty2024/states/nj/documents/nj1040_spec.rb

```diff
@@ -229,5 +229,55 @@
         end
       end
     end
+
+    describe "property tax - lines 40a and 40b" do
+      context "when taxpayer is a renter" do
+        let(:intake) { create(:state_file_nj_intake, :df_data_minimal, household_rent_own: 'rent', rent_paid: 54321) }
+
+        it "adds a checked tenant element to property tax deduct or credit" do
+          expect(xml.at("PropertyTaxDeductOrCredit Tenant").text).to eq("X")
+        end
+
+        it "does not add a checked homeowner element to property tax deduct or credit" do
+          expect(xml.at("PropertyTaxDeductOrCredit Homeowner")).to eq(nil)
+        end
+
+        it 'inserts property tax rent calculation on line 40a' do
```

Comment:
[DUST] Should we explain how we get to 9778 here like we do in the calculator? Seems confusing to have a different value here other than `rent_paid`.

Reply:
Hm. I wound up not explaining it here because the process that calculates that one isn't under test here, the way it is in the calculator. I used the same thinking in the XML builder test. Neither the XML nor PDF controls the conversion of one number to another, so I don't really want the tests to explain that conversion because it might imply that this is the class you'd want to look at to change that logic.

I used the word "calculation" in the test name to indicate that the value is derived from the calculator. Any ideas on how to make that clearer?

Reply:
Got it. After our conversation about TDD below, I think I agree. "Calculation" is probably sufficient here - alternative would be something like "inserts property tax rent amount from calculator on line 40a", but I don't think it's necessary.